src/init.cpp

```diff
@@ -1015,6 +1016,23 @@ bool AppInitParameterInteraction(const ArgsManager& args)
         }
     }
 
+    if (args.IsArgSet("-blockmaxweight")) {
+        const auto max_block_weight = args.GetIntArg("-blockmaxweight", DEFAULT_BLOCK_MAX_WEIGHT);
+        if (max_block_weight > DEFAULT_BLOCK_MAX_WEIGHT) {
+            return InitError(strprintf(_("Specified -blockmaxweight (%d) exceeds consensus maximum block weight (%d)"), max_block_weight, DEFAULT_BLOCK_MAX_WEIGHT));
+        }
+    }
+
+    if (args.IsArgSet("-blockreservedweight")) {
+        const auto block_reserved_weight = args.GetIntArg("-blockreservedweight", node::BlockCreateOptions{}.block_reserved_weight);
+        if (block_reserved_weight > DEFAULT_BLOCK_MAX_WEIGHT) {
```

Comment:
Why not check this against `max_block_weight` instead? The current behavior isn't catastrophic I guess because the clamping later silently fixes this. But it seems like the user is doing something wrong and in the case below we are throwing an error too instead of silently fixing it. Seems a little inconsistent at least.

```suggestion
        if (block_reserved_weight > max_block_weight) {
```

Reply:
My plan is to eventually remove the blockmaxweight option in a follow-up. See last sentence in [this comment](https://github.com/bitcoin/bitcoin/pull/31384#issuecomment-2567472713). The behavior you describe is an edge case that is possible, but as you said, the clamping fixes it.

I think it is consistent to compare using `DEFAULT_BLOCK_MAX_WEIGHT`  because, eventually, the block-reserved weight will be passed through the mining interface.

See the previous discussion on this [here](https://github.com/bitcoin/bitcoin/pull/31384#discussion_r1908522924).

Reply:
Indeed, this could be `MAX_BLOCK_WEIGHT`, but definitely not `max_block_weight`.

The clamping does not prevent someone from creating a 4MB coinbase transaction while otherwise not mining anything by setting `-maxblockweight=0`. As nonsensical as that might be, but dropping `-blockmaxweight` seems better to discuss in a followup.